packages/plugins/dart/flutter-freezed/src/config.ts

```diff
@@ -0,0 +1,522 @@
+import { CustomDecorator } from './utils';
+
+export interface FreezedConfig {
+  /**
+   * @name alwaysUseJsonKeyName
+   * @description Use @JsonKey(name: 'name') even if the name is already camelCased
+   * @default false
+   *
+   * @exampleMarkdown
+   * ```yml
+   * generates:
+   *   flutter_app/lib/data/models/app_models.dart
+   *     plugins:
+   *       - flutter-freezed
+   *     config:
+   *       alwaysUseJsonKeyName: true
+   *
+   * ```
+   */
+
+  alwaysUseJsonKeyName?: boolean;
+
+  /**
+   * @name assertNonNullableFields
+   * @description makes non-nullable fields optional properties in the Freezed class but uses Frezed's @Assert decorators to enforce them as required fields
+   * @default false
+   *
+   * @exampleMarkdown
+   * ```yml
+   * generates:
+   *   flutter_app/lib/data/models/app_models.dart
+   *     plugins:
+   *       - flutter-freezed
+   *     config:
+   *       assertNonNullableFields: true
+   *
+   * ```
+   */
+
+  assertNonNullableFields?: boolean; // TODO:
```

Comment:
what are those `TODO` standing for? is there still ongoing work on those properties?

Reply:
Those TODOs are just there to remind me to implement those config options as they add more features to the plugin.

Without them, the plugin works just fine. On that note, I am taking them out of the config in my next push

Reply:
@charlypoly  the presence or absence of those config options has a special meaning. They are used to decide which Freezed decorator to use. If even one of those options is set, then the plugin will use `@Freezed()` passing in the appropriate arguments to the `@Freezed` constructor. In their absence, the plugin fallback to using the normal `@freezed` decorator which accepts no configurations.

see this method

https://github.com/Parables/graphql-code-generator/blob/93e059d73e06ed98a074d4145100106e6a4f96fb/packages/plugins/dart/flutter-freezed/src/freezed-declaration-blocks/class-block.ts#L97

Reply:
@charlypoly  Removed, please check the latest commit